### ISSUE TITLE
Build failure - smart-default / quote

### ISSUE THREAD

**Issue by h1z1** [vi/websocat#29]

**Comment by vi** [comment]
Which Rust version do you use?

**Comment by vi** [comment]
Tried building 45523d81b0e08bda277616d80fff3b85e46c9f5c with `rustc 1.34.0-nightly (f6fac4225 2019-02-03)` and I don't see the error.

Also builds after `cargo update`. Also builds with `rustc 1.31.1 (b6c32da9b 2018-12-18)` and `rustc 1.32.0 (9fda7c223 2019-01-16)`, and `rustc 1.30.1 (1433507eb 2018-11-07)`, with or without `cargo update`.

**Comment by h1z1** [comment]
@vi  rustc 1.28.0 (9634041f0 2018-07-30)

**Comment by vi** [comment]
You should either update Rust or use Websocat version tagged as ["last_version_that_supports_rust_1_28"](https://github.com/vi/websocat/releases/tag/last_version_that_supports_rust_1_28). Currently you'll probably only miss the "--conncap" option compared to Websocat from master.

I noticed that README states 1.28 as a supported version. I'll update this stale information.

**Comment by h1z1** [comment]
Indeed.  Was 45523d81b0e08bda277616d80fff3b85e46c9f5c meant to detect that?

**Comment by vi** [comment]
This commit makes Travis check against 1.30 instead of 1.28. Also update dependencies that stopped supporting 1.28.

I want websocat to be eventually included in Debian repository, and outdated dependencies are not a good idea there. Rust 1.31 is Rust version in Debian Testing currently, so there's no real reason to support earlier Rust versions.

**Comment by h1z1** [comment]
> so there's no real reason to support earlier Rust versions.

Except every other distro / install that doesn't run nor track debian releases.  You're of course free to do as you wish, just noticed it didn't compile.

**Comment by vi** [comment]
Do you know any distros that are locked to older Rust versions? Debian is not a fast-paced distro and I expect it to have older Rust compared to a typical other distribution (other than Red Hat / CentOS)

If there are concrete examples of distributions that accept Rust-based packages into repositories and that are expected to be still 1.28 in half a year in future (for example), I may consider keeping a 1.28-compatible websocat and backport fixes there if possible.

**Comment by h1z1** [comment]
Coming from CentOS/RHEL, I wouldn't expect much in at least official repos if that is what you mean?  There are community backed ones of course.  Rust 1.31 is in EPEL.   1.29 isn't that old as far as api breaking deprecation goes but TBH I don't use Rust enough to care either way.     Wrote a wrapper in bash.

**Comment by vi** [comment]
Websocat 1.2.0 should be OK for current CentOS/RHEL then (I can try to make a RPM in Github releases if needed).

But Websocat 1.3.0 would probably require later Rust version.
